src/saml2/saml.py

```diff
@@ -51,41 +78,81 @@
     ('NAMEID_FORMAT_TRANSIENT', NAMEID_FORMAT_TRANSIENT),
     ('NAMEID_FORMAT_UNSPECIFIED', NAMEID_FORMAT_UNSPECIFIED),
 )
+
+# a profile outlines a set of rules describing how to embed SAML assertions.
+# https://docs.oasis-open.org/security/saml/v2.0/saml-profiles-2.0-os.pdf
+
+# XML based values for SAML attributes  
 PROFILE_ATTRIBUTE_BASIC = (
     "urn:oasis:names:tc:SAML:2.0:profiles:attribute:basic")
 
+# an AuthnRequest is made to initiate authentication
+# TODO: it is not clear that the request sets the context
+#       for the AuthnRequest, maybe rename to AUTHN_CONTEXT_PASSWORD
+
```

Comment:
This file has been auto-generated. We will leave it as it is for now.

There is planned work to replace those auto-generated files with new ones that support dataclasses with type-hinting along with a better parser and serializer.

```suggestion
```